src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -599,10 +655,14 @@ private boolean shouldRotateAndMaybeUpdateTimers(SinkRecord currentRecord, long
         "Should apply size-based rotation (count {} >= flush size {})? {}",
         recordCounter,
         flushSize,
-        messageSizeRotation
+        messageCountRotation
     );
 
-    return periodicRotation || scheduledRotation || messageSizeRotation;
+    String partitionEncoding = partitioner.encodePartition(currentRecord);
```

Comment:
`currentRecord` could be null when buffer is empty. This one throws NPE

Reply:
might explain why the build is failing